### PR TITLE
wallet, rpc: update listdescriptors response format

### PR DESCRIPTION
**Pull request by S3RK**
Update `listdescriptors` response format according to [RPC interface guidelines](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#rpc-interface-guidelines).

This is a follow up for #20226

**Before:**
```
Result:
[                               (json array) Response is an array of descriptor objects
  {                             (json object)
    "desc" : "str",             (string) Descriptor string representation
    "timestamp" : n,            (numeric) The creation time of the descriptor
    "active" : true|false,      (boolean) Activeness flag
    "internal" : true|false,    (boolean, optional) Whether this is internal or external descriptor; defined only for active descriptors
    "range" : [                 (json array, optional) Defined only for ranged descriptors
      n,                        (numeric) Range start inclusive
      n                         (numeric) Range end inclusive
    ],
    "next" : n                  (numeric, optional) The next index to generate addresses from; defined only for ranged descriptors
  },
  ...
]
```

**After:**
```
Result:
{                                 (json object)
  "wallet_name" : "str",          (string) Name of wallet this operation was performed on
  "descriptors" : [               (json array) Array of descriptor objects
    {                             (json object)
      "desc" : "str",             (string) Descriptor string representation
      "timestamp" : n,            (numeric) The creation time of the descriptor
      "active" : true|false,      (boolean) Activeness flag
      "internal" : true|false,    (boolean, optional) Whether this is internal or external descriptor; defined only for active descriptors
      "range" : [                 (json array, optional) Defined only for ranged descriptors
        n,                        (numeric) Range start inclusive
        n                         (numeric) Range end inclusive
      ],
      "next" : n                  (numeric, optional) The next index to generate addresses from; defined only for ranged descriptors
    },
    ...
  ]
}
```

### CODE REVIEW

**Review by meshcollider — COMMENTED** [review]
utACK 6c6803cdfd4889cfb9edccaaf4b6aeb081db2f94

File: src/wallet/rpcdump.cpp [hunk 1]
```diff
@@ -1761,8 +1762,8 @@ RPCHelpMan listdescriptors()
                     }},
                     {RPCResult::Type::NUM, "next", true, "The next index to generate addresses from; defined only for ranged descriptors"},
                 }},
-            }
-        },
+            }}
```
Comment (jonatack) [inline src/wallet/rpcdump.cpp:1765]: suggested fix line 1758

s/Whether this is internal or external descriptor/Whether this is an internal or external descriptor/

Comment (S3RK) [reply src/wallet/rpcdump.cpp:1765]: Fixed

**Comment by S3RK** [comment]
Fixed typo

**Comment by jonatack** [comment]
Tested ACK  94fe31eaf0dae8a970134bc0e6d7d69020a87daa

**Comment by achow101** [comment]
Code Review ACK 94fe31eaf0dae8a970134bc0e6d7d69020a87daa

**Comment by luke-jr** [comment]
I'm not sure it makes sense to return the `wallet_name` here... And the "prefer an Object result" advice (I assume this is trying to satisfy) doesn't seem like it should be applicable to list* methods.

Don't really care to argue against it if others disagree tho.

Concept ~0

**Review by meshcollider — COMMENTED** [review]
This requires rebase after https://github.com/bitcoin/bitcoin/commit/a69c3b35f8974b378a87a3e42d331bd4147e07df (#21277) due to changes in wallet_listdescriptors.py

**Review by meshcollider — COMMENTED** [review]
utACK 2e5f7def22e1b212fbd69e9147145d9d1f408aaf

**Comment by fanquake** [comment]
Concept ACK. @achow101 @jonatack want to take another look here?

**Comment by achow101** [comment]
re-ACK 2e5f7def22e1b212fbd69e9147145d9d1f408aaf

**Review by jonatack — COMMENTED** [review]
re-ACK 2e5f7def22e1b212fbd69e9147145d9d1f408aaf

new help
```
listdescriptors

List descriptors imported into a descriptor-enabled wallet.
Result:
{                                 (json object)
  "wallet_name" : "str",          (string) Name of wallet this operation was performed on
  "descriptors" : [               (json array) Array of descriptor objects
    {                             (json object)
      "desc" : "str",             (string) Descriptor string representation
      "timestamp" : n,            (numeric) The creation time of the descriptor
      "active" : true|false,      (boolean) Activeness flag
      "internal" : true|false,    (boolean, optional) Whether this is an internal or external descriptor; defined only for active descriptors
      "range" : [                 (json array, optional) Defined only for ranged descriptors
        n,                        (numeric) Range start inclusive
        n                         (numeric) Range end inclusive
      ],
      "next" : n                  (numeric, optional) The next index to generate addresses from; defined only for ranged descriptors
    },
    ...
  ]
}
```

**Pull request merged**